tools/clang/test/HLSLFileCheck/validation/verify_output_complete.hlsl

```diff
@@ -1,11 +1,9 @@
-// RUN: %dxc -T lib_6_8 %s | FileCheck %s
-
+// RUN: %dxc -Vd -T lib_6_8 %s | FileCheck %s
+// validation disabled to prevent the validator failing when (%dx.types.NodeRecordHandle zeroinitializer) gets generated
```

Comment:
The problem was that the validator prevented final dxil from being piped into filecheck, and so the Check Line would fail. 
The original purpose of this end-to-end test is to validate that re-using an output-completed handle should fail validation. With the new validation addition in this PR, it reveals an extra bug in the compiler (where zeroinitializer should be prevented from ever being a handle argument in the first place). I think this test has no purpose until this deeper issue is resolved, and then I can add a check line that shows the validator prevents reusing output-completed handles.

Reply:
> where zeroinitializer should be prevented from ever being a handle argument in the first place

This could be generated/discovered late in compilation, so I don't think we can always detect it through AST analysis.  An error later in the pipeline could be introduced to cause a failure before we complete compilation and pass it to the validator, but that may have diminishing returns.  Detection of clear cases at the AST level should be performed for better diagnostics, which should easily be able to catch this particular case.

Since the IR test that's part of this PR already checks the validation for zeroinit/undef handle usage, we can probably remove this test from this PR, as long as we have an issue tracking the work item to detect the case in the AST.  Then we can attach this example to that work item, since this test could be converted to a verifier test at that point.